test/replaytests/workflows.go

```diff
@@ -113,3 +113,13 @@ func helloworldActivity(ctx context.Context, name string) (string, error) {
 	logger.Info("helloworld activity started")
 	return "Hello " + name + "!", nil
 }
+
+// TimerWf starts a timer and always starts another timer at workflow end, even if cancelled
+func TimerWf(ctx workflow.Context) error {
+	defer func() {
+		// Produce another timer after being cancelled
+		newCtx, _ := workflow.NewDisconnectedContext(ctx)
```

Comment:
You could do defer func() { workflow.ExecuteActivity() } and we would run into the same problem, right? I don't think this fix is good enough. We need stop sending more command when context is cancelled/errored.

Reply:
I misunderstand the problem. Sending cancellation command in response to workflow cancellation is fine and expected. The problem is sending cancellation command when the corresponding timer/activity etc is closed in the same workflow task. I think the fix is reasonable.

Reply:
Yeah that's very much what I say in the description and mentioned in the meeting. We know this fix doesn't really work properly, but it's also just not possible to fix without huge changes. Simply not sending more commands isn't going to work, because we still want to issue cancellations for things that *aren't* also resolved in the same workflow task.

Buffering them and sending them at the end of the task works, but causes backwards incompatible changes to the way history is created, which we talked about in the meeting too.

So unless there's a simpler fix here which wasn't on the table this cheap patch is what we agreed we'd be fine with, but if you've got an idea of a better way I'm all ears.